development/src/logic/pieces.ts

```diff
@@ -78,8 +78,36 @@ export type Square = {
 };
 
 export class Pawn extends Piece {
+
+  enPassant: boolean;
+
   constructor(position: Position, player: Player) {
     super(position, player, pawnResource, 'Pawn');
+    this.enPassant = false;
+  }
+
+  enPassantCheck(targetPosition: Position, draggedPiece: Piece){
+
+    let changeInPosition = 0;
+  
+    if (targetPosition.coordinates[1] === 2) {
+      changeInPosition = 1;
+    }
+    else if (targetPosition.coordinates[1] === 5){
+      changeInPosition = -1;
+    } else return false;
+  
+    targetPosition.coordinates[1] += changeInPosition;
+    const targetPiece = getPieceByPositionAndBoard(targetPosition);
+    if ((targetPiece instanceof Pawn) && targetPiece.enPassant){
+      actOnTurnPieceToPiece(draggedPiece,targetPiece, false);
+      targetPosition.coordinates[1] -= changeInPosition;
```

Comment:
Since you are moving a piece here, shouldn't this function be in the logic script?